src/components/SpaceDelegatesProfileModal.vue

```diff
@@ -0,0 +1,173 @@
+
+const { web3Account } = useWeb3();
+const { formatCompactNumber } = useIntl();
+const { fetchDelegate, delegate } = useDelegates(props.space.delegationPortal);
+const { domain } = useApp();
+const {
+  loadingStatements,
+  loadStatements,
+  reloadStatement,
+  getStatementAbout,
+  getStatementStatement,
+  formatPercentageNumber
+} = useStatement();
+
+const showModalStatement = ref(false);
+
+const isLoggedUser = computed(() => {
+  return web3Account.value.toLowerCase() === props.address.toLowerCase();
+});
+
+watch(
+  () => props.address,
+  async () => {
+    fetchDelegate(props.address);
+    loadStatements(props.space.id, [props.address]);
+  }
+);
+</script>
+
+<template>
+  <BaseModal :open="open" @close="emit('close')">
+    <div class="p-4">
+      <div class="flex">
+        <div>
+          <AvatarUser :address="address" size="69" />
+        </div>
+        <div>
+          <ProfileName :profile="profiles[address]" :address="address" />
+          <div class="flex gap-3 pl-3 text-skin-text">
+            <template v-if="delegate">
+              <div
+                v-tippy="{
+                  content: formatPercentageNumber(delegate.votesPercentage)
+                }"
+              >
+                {{ formatCompactNumber(Number(delegate.delegatedVotes)) }}
+                {{ space.symbol }}
+              </div>
+              <div
+                v-tippy="{
+                  content: formatPercentageNumber(delegate.delegatorsPercentage)
+                }"
+              >
+                {{
+                  formatCompactNumber(
+                    Number(delegate.tokenHoldersRepresentedAmount)
+                  )
+                }}
+                delegators
+              </div>
+            </template>
+          </div>
+        </div>
+      </div>
+
+      <LoadingList v-if="loadingStatements" class="mt-4" />
+      <div v-else class="mt-4 space-y-3">
+        <div v-if="getStatementAbout(address)">
+          <div class="flex items-center">
+            About
+            <BaseButtonIcon v-if="isLoggedUser">
+              <i-ho-pencil
+                class="text-xs"
+                @click="
+                  emit('close');
+                  showModalStatement = true;
+                "
+              />
+            </BaseButtonIcon>
+          </div>
+          <p class="text-skin-heading">
+            {{ getStatementAbout(address) }}
+          </p>
+        </div>
+
+        <div v-if="getStatementStatement(address)">
+          <div class="flex items-center">
+            Statement
+            <BaseButtonIcon v-if="isLoggedUser">
+              <i-ho-pencil
+                class="text-xs"
+                @click="
+                  emit('close');
+                  showModalStatement = true;
+                "
+              />
+            </BaseButtonIcon>
+          </div>
+          <p class="text-skin-heading">
+            <BaseMarkdown :body="getStatementStatement(address)!" />
```

Comment:
This is new, what will happen if we add `!` after the function 🤔 is it vue specific?

Reply:
https://stackoverflow.com/questions/42273853/in-typescript-what-is-the-exclamation-mark-bang-operator-when-dereferenci

Reply:
There is a check above `<div v-if="getStatementStatement(address)">` but typescript doesn't pick that up for some reason. The `!` is just telling typescript that it is always `non-null`

Reply:
Oh i see, i get this error from `eslint-plugin-vue` so i asked

```
[vue/no-parsing-error]
Parsing error: Unexpected token !.eslint-plugin-vue
```